hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -320,6 +327,27 @@ private void saveInternalSchema(HoodieTable table, String instantTime, HoodieCom
     }
   }
 
+  protected HoodieTable createTableAndValidate(HoodieWriteConfig writeConfig,
```

Comment:
similar all table write operations call this to validate version, before beginning any operation.